### ISSUE TITLE
ProjectHarvesting DocumentOutputGroup does not use the correct configuration when building with MSBuild

### ISSUE THREAD

**Issue by jmairboeck** [wixtoolset/issues#7537]
* Which version of WiX are you building with?

> 4.0.1

* Which version of Visual Studio are you building with (if any)?

> 17.6.2

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?

> HeatWave 1.0.1.1

* Which version of .NET are you building with?

> .NET Framework 4.8

* If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?

> N/A

* Describe the problem and the steps to reproduce it.

When harvesting (SDK-style) projects which contain something like

```xml
  <PropertyGroup>
    <DocumentationFile>bin\$(Configuration)\MyProject.XML</DocumentationFile>
  </PropertyGroup>
```

this is harvested always as `bin\Debug\MyProject.XML` in the DocumentOutputGroup, even if the built configuration is not Debug (the default configuration). This causes the MSI package build to fail because the file is not found if the Debug configuration has not been built.

* Describe the behavior you expected and how it differed from the actual behavior.

> The actually built Configuration should be used when harvesting files in the DocumentationOutputGroup, not just "Debug".

**Comment by barnson** [comment]
Please provide a simple repro project in a GitHub repo.

**Comment by jmairboeck** [comment]
The concrete examples which fail are:

https://github.com/VPKSoft/Scintilla.NET/blob/master/Scintilla.NET/Scintilla.NET.csproj

and

https://github.com/VPKSoft/ScintillaNET-FindReplaceDialog/blob/master/ScintillaNet_FindReplaceDialog/ScintillaNET_FindReplaceDialog.csproj

**Comment by jmairboeck** [comment]
We are not actually using the documentation files, but they are harvested anyway and only later not included in the final package using `%(ProjectReference.RefProjectOutputGroups)`.

**Comment by barnson** [comment]
Any .wixprojs?

**Comment by jmairboeck** [comment]
Not publicly unfortunately. I would have to build a simplified version of our setup and publish it. I will do that next week probably.

**Comment by jmairboeck** [comment]
I just noticed that this only happens when building the .wixproj on the command line with MSBuild (or using the `MSBuild` MSBuild task, like our deployment script does) and not when building in Visual Studio. I didn't try `dotnet build`.

However, that way, the project references don't get any configuration assigned and therefore default to "Debug". So this issue can probably be closed as invalid.

Interestingly though, the TargetPath of the ProjectReference gets set according to the configuration of the .wixproj that is being built.

Additionally, the .wixproj where we do the harvesting is a library, so we apply the workaround from #3496, which is easily adjusted to work around this issue (by not harvesting Documents).

**Comment by jmairboeck** [comment]
This is not the same as #3496. I just closed it as invalid because I realized that it only happens when just building the .wixproj standalone without the solution context. In this case, nothing actually specifies which configuration to use for the ProjectReferences, so this can't really work.

It just happened to be the case that the project in question is a wixlib project, so this issue was easily worked around by adjusting the original workaround from #3496. But these are two different aspects.
